lib/puppet/type/ssh_authorized_key.rb

```diff
@@ -48,6 +48,23 @@ module Puppet
       isnamevar
     end
 
+    newparam(:drop_privileges) do
+      desc "Whether to drop privileges when writing the key file. This is
+      useful for creating files in paths not writable by the target user. Note
+      the possible security implications of managing file ownership and
+      permissions as a privileged user."
+
+      munge do |value|
+        value.to_s == 'true'
```

Comment:
@joshcooper is there a better way to ensure we ultimately get boolean values for this parameter?

I added this because when managing a resource with `puppet resource ... drop_privileges=false`, `@resource[:drop_privileges]` becomes a string.

Reply:
I tried setting `:parent => Puppet::Parameter::Boolean` but it gives me errors when I manage resources with `puppet resource`. Manually applying puppet manifests works though.

Reply:
boolean parameters need to be defined like the [user's `managehome` parameter](https://github.com/puppetlabs/puppet/blob/69e81261559b8f3327bea55c34382acda5511068/lib/puppet/type/user.rb#L404)

```
      newparam(:managehome, :boolean => true, :parent => Puppet::Parameter::Boolean) do
```

The parent boolean type should coerce the string to a boolean value as part of the `munge` process.

Reply:
Boolean parameters are pretty messed up in puppet. Check out https://tickets.puppetlabs.com/browse/PUP-8442?focusedCommentId=537367&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-537367

Reply:
I'll try that again, I think I just forgot to require `puppet/parameter/boolean` in the type so that's why it didn't work.